presto-main-base/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -298,6 +301,44 @@ private RelationPlan addColumnMasks(Table table, RelationPlan plan, SqlPlannerCo
         return new RelationPlan(planBuilder.getRoot(), plan.getScope(), newMappings.build());
     }
 
+    @Override
+    protected RelationPlan visitTableFunctionInvocation(TableFunctionInvocation node, SqlPlannerContext context)
+    {
+        Analysis.TableFunctionInvocationAnalysis functionAnalysis = analysis.getTableFunctionAnalysis(node);
+
+        // TODO handle input relations:
```

Comment:
its not clear yo me but looks like only scalar arguments work and table within table won't so can we fail gracefully for unsupported cases?